makefile

```diff
@@ -70,5 +70,15 @@ getdependency:
 #######################################
 # compile all tests under $(TEST_ROOT)
 #######################################
+log=$(TESTOUTPUT)$(D)compile$(D)compilation.log
```

Comment:
please rename `log` to be more specific, something like `COMPILATION_LOG` will be better